apps/root/src/pages/strategy-guardian-detail/frame/index.tsx

```diff
@@ -72,6 +74,16 @@ const StrategyDetailFrame = () => {
     }
   }, [chainId, strategyGuardianId, sdkMappedNetworks]);
 
+  React.useEffect(() => {
+    if (strategyGuardianId) {
+      try {
+        void earnService.fetchMultipleEarnPositionsFromStrategy(strategyGuardianId);
+      } catch (error) {
+        console.error('Failed to fetch detailed strategy', chainId, strategyGuardianId, error);
+      }
+    }
+  }, [strategyGuardianId, earnPositions]);
```

Comment:
Why should this be updated whenever a user position changes? Or is this just to trigger the `needsToUpdateUserStrategy` check?

Reply:
This is to be triggered assuming earnpositions are already on the service and to update their data, since we are going to need a lot of them and didn't want us to be overriding stuff on the go